kyaml/kio/kioutil/kioutil.go

```diff
@@ -44,18 +44,27 @@ const (
 )
 
 func GetFileAnnotations(rn *yaml.RNode) (string, string, error) {
-	if err := CopyLegacyAnnotations(rn); err != nil {
-		return "", "", err
+	annotations := rn.GetAnnotations()
+	path, found := annotations[PathAnnotation]
+	if !found {
+		path = annotations[LegacyPathAnnotation]
 	}
-	meta, err := rn.GetMeta()
-	if err != nil {
-		return "", "", err
+	index, found := annotations[IndexAnnotation]
+	if !found {
+		index = annotations[LegacyIndexAnnotation]
```

Comment:
Are there any cases where both of these might be set? Is preferring the internal annotation over the legacy one always the right thing to do?

Reply:
> Are there any cases where both of these might be set?

Yes

> Is preferring the internal annotation over the legacy one always the right thing to do?

Not sure. But we can add godoc for it.

I no longer uses `CopyLegacyAnnotations` here since IMO it's unexpected that a Get function actually modify the object.
The behavior is identical to the it used to be except there are no modification:
- If legacy annotation is empty, we get the value from the new one.
- If new annotation is empty, we get the value from the legacy one.
- If neither is empty, we use the new one.
- If both are empty, we get empty value.